site/content/how-to/traffic-management/snippets.md

```diff
@@ -0,0 +1,390 @@
+      Message:               SnippetsFilter is accepted
+      Observed Generation:   1
+      Reason:                Accepted
+      Status:                True
+      Type:                  Accepted
+    Controller Name:         gateway.nginx.org/nginx-gateway-controller
+Events:                      <none>
+```
+
+
+Update the tea HTTPRoute to reference the `SnippetsFilter`:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: tea
+spec:
+  parentRefs:
+    - name: gateway
+      sectionName: http
+  hostnames:
+    - "cafe.example.com"
+  rules:
+    - matches:
+        - path:
+            type: PathPrefix
+            value: /tea
+      filters:
+        - type: ExtensionRef
+          extensionRef:
+            group: gateway.nginx.org
+            kind: SnippetsFilter
+            name: tea-rate-limiting-sf
+      backendRefs:
+        - name: tea
+          port: 80
+EOF
+```
+
+Verify that the tea HTTPRoute's has been configured correctly:
+
+```shell
+kubectl describe httproutes.gateway.networking.k8s.io tea
+```
+
+You should see the following conditions:
+
+```text
+Conditions:
+      Last Transition Time:  2024-10-28T00:33:08Z
+      Message:               The route is accepted
+      Observed Generation:   2
+      Reason:                Accepted
+      Status:                True
+      Type:                  Accepted
+      Last Transition Time:  2024-10-28T00:33:08Z
+      Message:               All references are resolved
+      Observed Generation:   2
+      Reason:                ResolvedRefs
+      Status:                True
+      Type:                  ResolvedRefs
+```
+
+Test that the `SnippetsFilter` is configured and has successfully applied the rate limiting NGINX configuration changes.
+
+Send a request to tea:
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/tea
+```
+
+This request should receive a response from the tea Pod:
+
+```text
+Server address: 10.244.0.6:8080
+Server name: tea-76c7c85bbd-cf8nz
+```
+
+When processing a single request, the rate limiting configuration has no noticeable effect. Try sending
+multiple requests.
+
+```shell
+for i in `seq 1 10`; do curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/tea; done
+```
+
+You should see all successful responses from the tea Pod, but they should be spaced apart roughly one second each as
+expected through the rate limiting configuration.
+
+You've now used the `SnippetsFilter` resource to configure two distinct rate limiting rules to different backend applications through the use of Snippets.
+For an alternative method of modifying the NGINX configuration NGINX Gateway Fabric generates through Gateway API resources, check out
+our supported [first-class policies]({{< relref "overview/custom-policies.md" >}}) which don't carry many of the aforementioned disadvantages of Snippets.
+
+---
+
+## Further reading
+
```

Comment:
I think we need a section here or in our troubleshooting guide about how to diagnose issues with SnippetsFilters. Most importantly, we need to mention somewhere (probably here) that if the SnippetsFilter cannot be resolved, the route will return a 500 HTTP error code response.

Reply:
Added a troubleshooting section in the document, let me know if you think there's anything that needs to be adjusted or added/removed, thanks!

Reply:
I think we should move this to [troubleshooting guide ](https://github.com/nginxinc/nginx-gateway-fabric/blob/main/site/content/how-to/monitoring/troubleshooting.md). Its better consolidated in one place and we don't generally follow the practice of having troubleshooting sections with our `traffic-management` guides

Reply:
True, though I did see https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-snippets/ and how they put a troubleshooting guide in with the snippets that is separate from all of their other troubleshooting guides. this also breaks their practices too it seems (I just checked some of their guides and they don't have troubleshooting guides in them) 

@ADubhlaoich thoughts?

Reply:
Really great troubleshooting section!

I personally like the troubleshooting info to be close to the documentation on the feature, but I'll defer to @ADubhlaoich.

Reply:
I like the in-page troubleshooting section, since if anything goes wrong during set-up, it'll be close to where they need it.

As a design pattern we're trying to slowly move troubleshooting information to the pages where it's relevant: standalone troubleshooting pages can occasionally feel like the equivalent of a FAQ. If you need one, it's a sign information might be unclear elsewhere.

General guidance on how to troubleshoot or debug things remains permissible on their own pages (Such as enabling or configuring logging, as an example).